src/Chainweb/TreeDB/Difficulty.hs

```diff
@@ -61,14 +61,23 @@ hashTarget db bh
         let delta :: TimeSpan Int64
             !delta = TimeSpan $ time bh' - time start
 
-        pure . adjust ver delta $ _blockTarget bh'
+        pure . adjust ver (WindowWidth ww) delta $ _blockTarget bh'
   where
+    height :: Natural
+    height = int $ _blockHeight bh'
+
     bh' :: BlockHeader
     bh' = bh ^. isoBH
 
     ww :: Natural
     ww = case window ver of
-      Just (WindowWidth n) -> n
+      Just (WindowWidth n)
+          -- Intent: Initial network conditions are chaotic. `HashTarget`s are
+          -- so easy that mining progresses extremely quickly. In order to not
+          -- overwhelm the Cut network, we perform earlier adjustments before
+          -- the first "official" adjustment epoch is reached.
+          | height < n -> n `div` 10
+          | otherwise -> n
```

Comment:
If we derive `Num` for `WindowWidth`, alot of this unwrapping and rewrapping goes away, ya?

Reply:
I second this!

Reply:
I'll give it a shot.